text/0001-zap-process.md

```diff
@@ -0,0 +1,106 @@
+may choose to open a Zulip architecture proposal (ZAP).
+
+A ZAP is a pull request in this `zulip-architecture` repository that adds a
+Markdown document named like `text/0000-my-feature.md`. The numerical ID should
+be initialized with the placeholder `0000`, and replaced with the pull request
+number after the pull request is opened.
+
+The pull request description should link to the rendered Markdown document from
+the author's branch. The Markdown document should begin with a link back to the
+GitHub pull request page, and proceed with a suggested organization into
+sections such as “Summary”, “Motivation”, “Detailed design”, “Alternatives”,
+“Security implications”, “Future directions”, and “Previous discussion”, as
+applicable.
+
+Each new ZAP pull request should be posted to an appropriate stream in the
+development community for discussion. Discussion can happen in the development
+community or in comments/reviews on the pull request. Any discussion that
+happens outside the pull request should ideally be linked and summarized on the
+pull request to keep a record of it. The pull request will remain open while
+discussion is ongoing, and may be updated as desired.
+
+If the Zulip team reaches a consensus (via the usual decision-making conventions
+of the Zulip project, which are deliberately left unspecified here) that the
+proposal should be accepted, the pull request will be merged. If consensus is
+that the proposal should be rejected, the pull request will be closed, and if
+consensus is that it needs modifications, the proposal may remain open while
+being edited, or be closed in favor of a different pull request.
+
+The documents in `zulip-architecture` will be maintained to reflect the current
+consensus of the project. (This maintenance may involve using issues and pull
+requests that update previously accepted ZAPs but are not themselves ZAPs, so
+ZAP numbers may be nonconsecutive.)
+
+# Alternatives
+
+Most design decisions in Zulip have been made in the
+[development community](https://zulip.com/development-community/) or within the
+GitHub pull requests for the implementation of individual Zulip features. We
+expect that most smaller decisions will continue to be made in those ways, and
+the ZAP process will be used in planning more substantial changes that will
+benefit from it.
+
+Some design proposals have been drafted on other platforms such as Dropbox
+Paper, but these are not accessible to our open source contributors outside the
+core team and aren't collected in a useful way.
+
+We considered writing design proposals within the Zulip
+[developer documentation](https://zulip.readthedocs.io/) which lives in the same
+repository as the server code. However, the developer documentation is intended
+to document Zulip as currently implemented; it has different maintenance and
+lifecycle needs from proposals that may pertain to a future or past
+implementation changes.
```

Comment:
Maybe add a sentence of the form "Where appropriate, content can be copied from a ZAP into the developer documentation once the ZAP is implemented; but rarely will it make sense for a ZAP document to be copied in its entirely, since the audience and goals are different."